### PR TITLE
Feature/improve sorting ux #1541

### PR DESCRIPTION
**Pull request by kaperoo**
## Improved sorting UX
This PR implements improved sorting UX.
Changes/ New features:
- Each sortable (but not sorted) column now has an 'SortIcon' icon so users know it is clickable. 
- When shift key is pressed (for multi sort), all Sort Icons change to Add Icons  

- Single column sort is now the default
single sort

- Multi column sort is applied when columns are clicked with shift key down

- Sorting in the card view works in a similar way to table view, except it doesn't display 'SortIcon' (please let me know if it should). 

## Testing instructions
Had to modify multiple existing unit and e2e tests to account for new way of multi column sorting.
Additionally, added tests checking for icon changes.

## Agile board tracking
connect to #1541

### CODE REVIEW

**Comment by louise-davies** [comment]
@kevinphippsstfc do you have any thoughts about the sort icon?

File: packages/datagateway-common/src/card/cardView.component.tsx
Comment (louise-davies) [inline packages/datagateway-common/src/card/cardView.component.tsx:262]: Minor comment, but since you specified `shiftDown?: boolean` as optional, you could leave out explicitly specifying `false` here. Would have meant less changes in the tests - but no point removing it now, just an observation

File: packages/datagateway-common/src/card/cardView.component.tsx
Comment (louise-davies) [inline packages/datagateway-common/src/card/cardView.component.tsx:663]: You could simply & reduce code repetition here by removing the ternary checks and changing `TableSortLabel` to:

```
<TableSortLabel
   active={s.dataKey in sort}
   direction={sort[s.dataKey]}
   // Set tabindex to -1 to prevent button focus
   tabIndex={-1}
   IconComponent={!(s.dataKey in sort) && shiftDown ? AddIcon : undefined}
>
   {s.dataKey in sort && sort[s.dataKey]}
</TableSortLabel>
```

File: packages/datagateway-common/src/table/headerRenderers/dataHeader.component.tsx
Comment (louise-davies) [inline packages/datagateway-common/src/table/headerRenderers/dataHeader.component.tsx:103]: Again, some code repetition here. We can reduce this by only adding the extra props when `dataKey in sort` is false, e.g.

```
<TableSortLabel
  className={'tour-dataview-sort'}
  active={true}
  direction={!(dataKey in sort) ? 'desc' : currSortDirection}
  onClick={(event) => {
    onSort(dataKey, nextSortDirection, 'push', event.shiftKey);
    if (!(dataKey in sort)) setHover(false);
  }}
  {...(!(dataKey in sort)
    ? {
        onMouseEnter: () => setHover(true),
        onMouseLeave: () => setHover(false),
        IconComponent: hover
          ? ArrowUpwardIcon
          : shiftDown
          ? AddIcon
          : SortIcon,
        sx: {
          transition: 'opacity 0.5s',
          opacity: hover ? 0.7 : 1,
        },
      }
    : {})}
>
  <Typography noWrap sx={{ fontSize: 'inherit', lineHeight: 'inherit' }}>
    {label}
  </Typography>
</TableSortLabel>
```

**Review by louise-davies — COMMENTED** [review]
Works well, just some minor code styling comments

**Review by louise-davies — COMMENTED** [review]
Comments have been addressed - just want to wait for @kevinphippsstfc to take a look before we merge

**Review by louise-davies — CHANGES_REQUESTED** [review]
Kevin suggested we add hover tooltips to the sort indicators to be explicit that they're prompting the users to sort :)

**Comment by kaperoo** [comment]
Added requested hover tooltips to sort indicators.

When hovered:
image

When hovered with shift down:
image

Let me know if the tooltips should be more descriptive (i.e. "Sort asc") :)

**Review by louise-davies — APPROVED** [review]
LGTM and Kevin was happy with the UI & the tooltips are working well

**Pull request merged**
